src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1119,50 +1057,82 @@ void CASESession::SendErrorMsg(SigmaErrorType errorCode)
                    ChipLogError(SecureChannel, "Failed to send error message"));
 }
 
-CHIP_ERROR CASESession::FindValidTrustedRoot(const System::PacketBufferTLVReader & tlvReader, uint32_t nTrustedRoots)
+CHIP_ERROR CASESession::GenerateDestinationID(const ByteSpan & random, const Credentials::CertificateKeyId * trustedRootId,
```

Comment:
Suggest making this take in just the raw public key of the trusted root, rather than needing to get access to state (`mOpCredSet`) from within the primitive. That would make testing easier.

Reply:
Updated.